poems_parser.py

```diff
@@ -1,67 +1,92 @@
 def get_resource(url):
     resource = urllib.request.urlopen(url)
     return resource.read()
 
 
-# Parse html
-def parse_page(html):
+# parser
+def parse_page(html, url):
     res = []
     soup = BeautifulSoup(html, 'html.parser')
     table = soup.find('table', width="99%", border="0", cellspacing="2", cellpadding="2")
     for row in table.find_all('td', valign='top'):
-        res.append(row.find('a', class_='linki2'))
-    return res
+        poem_name = row.find('a', class_='linki2').text
+        poem_ur = url + row.find('a', class_='linki2')['href']
+        try:
+            poem_year = row.find('span').text
+        except AttributeError:
+            continue
+        poem_year = re.findall('[0-9]{4}', poem_year)
 
+        res.append([poem_name, poem_year[0], poem_ur])
+    return res
 
-def main():
-    # parse the arguments
-    parser = argparse.ArgumentParser(description='My arsg parser')  # You can add better description :)
-    parser.add_argument('--output_file', action="store", type=str)
-    args = parser.parse_args()
-    # parse the poems titles
-    poem = parse_page(get_resource('http://taras-shevchenko.in.ua/virshi-shevchenka.html'))
-    # create folder
-    path = os.path.dirname(os.path.abspath(args.output_file))
-    if not os.path.exists(path):   # in python3.7+ you can avoid this check, just:
-        os.makedirs(path)          # os.makedirs(path, exist_ok=True)
 
-    # There is a way to make it much more simple
-    # with open(args.output_file, "w") as f:
-    #     for p in poem:
-    #         f.write(p.string.strip() + '\n')
+# парсить посилання на вірш
+def parse_poem_by_url(url):
+    poem_page = get_resource(url)
+    soup = BeautifulSoup(poem_page, 'html.parser')
+    poem_header = soup.find('span', class_="text-head2").text
+    poem_text = soup.find('p', align='left').text
+    poem = (poem_header + '\n' + poem_text + '\n')
+    return poem
 
-    # open file
-    file_to_write = codecs.open(args.output_file, "w", "utf-8")
-    # write to file
-    for p in poem:
-        file_to_write.write(p.string.strip() + '\n')
-    # close the file
-    file_to_write.close()
 
-    # you don't need read this data from file, just:
-    # print('Amount of poems =', len(poem))
+def print_whole_poem(poem_url):
+    poem = parse_poem_by_url(poem_url)
+    print(poem)
 
-    with codecs.open(args.output_file, "r", "utf-8") as file:
-        print('Amount of poems =', len(file.readlines()))
 
-    print("Absolute path = ", path)
+def main():
+    # parse the poems titles
+    url = 'http://taras-shevchenko.in.ua/'
+    poem = parse_page(get_resource(url + 'virshi-shevchenka.html'), url)
+    # parse the arguments
+    parser = argparse.ArgumentParser(description='My arsg parser')
+    parser.add_argument('--output_file', action="store", type=str)
+    parser.add_argument('--sort_poems', '-name', '-years', action='store', nargs=1)
+    parser.add_argument('--sort_order', '-asc', '-desc', action='store', nargs=1)
```

Comment:
Default argument is usefull, in this case you'll have behaviour for the  situation when this arg missed
```suggestion
    parser.add_argument('--sort_order', choices=['asc', 'desc'], default='asc')
```